common/src/main/java/com/wynntils/models/gear/SetModel.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.gear;
+
+import com.google.gson.reflect.TypeToken;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.gear.type.GearSlot;
+import com.wynntils.models.gear.type.SetInfo;
+import com.wynntils.models.gear.type.SetInstance;
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.models.items.items.game.GearItem;
+import com.wynntils.models.stats.type.StatType;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.mc.McUtils;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.item.ItemStack;
+
+public class SetModel extends Model {
+    public static final Pattern SET_PATTERN = Pattern.compile("§a(.+) Set §7\\((\\d)/\\d\\)");
+
+    // Stored as a map for quick lookup <name, SetInfo>
+    private final Map<String, SetInfo> setData = new HashMap<>();
+    private final Map<GearSlot, SetInstance> setInstances = new EnumMap<>(GearSlot.class);
+
+    public SetModel() {
+        super(List.of());
+        loadSetData();
+    }
+
+    public SetInfo getSetInfo(String setId) {
+        return setData.getOrDefault(setId, null);
+    }
+
+    public SetInfo getSetInfoForItem(String itemName) {
+        return getSetInfo(getSetName(itemName));
+    }
+
+    /**
+     * @param itemName The name of the item to check
+     * @return The set name if the item is part of a set, null otherwise
+     */
+    public String getSetName(String itemName) {
+        for (Map.Entry<String, SetInfo> entry : setData.entrySet()) {
+            if (entry.getValue().items().contains(itemName)) {
+                return entry.getKey();
+            }
+        }
+        return null;
+    }
+
+    public void updateSetInstance(GearSlot slot, SetInstance instance) {
+        setInstances.put(slot, instance);
+    }
```

Comment:
What if we just didn't save this data in the model? It is a weird dependency, having the annotator call this model.

Do you see anything wrong with just storing the set instance in the GearItem, and just collecting all worn items and checking them in `getTrueCount`?

Reply:
Also, you basically don't even use this map anymore from what I can see. Collecting the unique worn sets can be done by collecting it from the gearItems. (to clarify, `getUniqueSetNames` should stay, but should be modified).

Reply:
Also, with the map removed, `GearSlot` itself can go.